api/src/pcapi/core/offers/api.py

```diff
@@ -1485,3 +1487,24 @@ def edit_shadow_stock(stock: Stock, stock_data: dict) -> Stock:
     search.async_index_offer_ids([stock.offerId])
 
     return stock
+
+
+def update_stock_quantity_to_match_booking_provider_remaining_place(offer: Offer) -> None:
+    booking_provider = VenueBookingProvider.query.filter(
+        VenueBookingProvider.venueId == offer.venueId, VenueBookingProvider.isActive
+    ).one_or_none()
+
+    if booking_provider:
+        shows_id = []
+        for stock in offer.activeStocks:
+            if stock.idAtProviders and stock.idAtProviders.isdigit():
+                shows_id.append(int(stock.idAtProviders))
+        shows_remaining_places = get_shows_stock(offer.venueId, shows_id)
+
+        for show_id, remaining_places in shows_remaining_places.items():
+            stock = next((s for s in offer.activeStocks if s.idAtProviders == str(show_id)))
+            if stock:
+                if remaining_places <= 0:
+                    stock_to_update = offers_repository.get_and_lock_stock(stock.id)
+                    stock_to_update.quantity = 0
```

Comment:
```suggestion
                    stock_to_update.quantity = stock_to_update.dnBookedQuantity
```
La quantité du stock ne peut pas être inférieur au nombre de réservations sur le stock en question. Au lieu de mettre 0 on mettra le nombre de réservation et ça implique un stock épuisé

Reply:
Je ne suis pas sur d'avoir compris cette partie si on prend le cas d'un stock avec une quantité initiale à 10 et qu'il y a 7 réservations de faites sur ce stock, le stock aura une quantité de 3 non ? (donc inférieur au nombre de résas sur ce stock je me trompe ?)

Reply:
En fait le champs `quantity` dans stock correspond toujours à la quantité initiale, il n'est pas modifie en fonction de réservation. C'est plutôt le champs `dnBookedQuantity` qui contient le nombre de réservations sur un stock. Donc la quantité restante sur un stock = `quantity` - `dnBookedQuantity`

Reply:
Ah ok beaucoup plus clair du coup, merci !